platform/packages/platform/src/reply.rs

```diff
@@ -23,78 +25,85 @@ impl InstantiateResponse<Vec<u8>> {
     }
 }
 
-pub fn from_instantiate_addr_only(api: &dyn Api, reply: Reply) -> StdResult<Addr> {
+pub fn from_instantiate_addr_only(api: &dyn Api, reply: Reply) -> Result<Addr> {
     from_instantiate_inner::<MsgInstantiateContractResponse>(api, reply)
         .map(InstantiateResponse::into_addr)
 }
 
-pub fn from_instantiate2_addr_only(api: &dyn Api, reply: Reply) -> StdResult<Addr> {
+pub fn from_instantiate2_addr_only(api: &dyn Api, reply: Reply) -> Result<Addr> {
     from_instantiate_inner::<MsgInstantiateContract2Response>(api, reply)
         .map(InstantiateResponse::into_addr)
 }
 
-pub fn from_execute<T>(reply: Reply) -> StdResult<Option<T>>
+pub fn from_execute<T>(reply: Reply) -> Result<Option<T>>
 where
     T: DeserializeOwned,
 {
     decode::<MsgExecuteContractResponse>(reply)
         .map(|data| data.data)
-        .map(Binary)
-        .and_then(from_json)
+        .and_then(|data| from_json(Binary(data)).map_err(From::from))
```

Comment:
no need to instantiate `Binary`, just use `&data`

Reply:
Must be too scarred from the old API. Didn't even even notice. Thanks!